pemFioi/quickAlgo/interface-mobileFirst.js

```diff
@@ -302,7 +308,78 @@ var quickAlgoInterface = {
+    },
+
+    shareSubject: function() {
+        // we take back quickAlgoInterface from global variable because the word "this" refer to displayHelper
+        var that = quickAlgoInterface;
+
+        var additional = that.userTaskData;
+
+        // This is done as if it was a task.js
+        var subTaskToPublish = {
+            // the title of the exercise
+            // we will have to move this to userTaskData during loading.
+            title: additional.title,
+
+            // taskIntro is the subject inside of the json editor.
+            // we will have to move this to userTaskData during loading.
+            taskIntro: additional.subject,
+
+            // This are the meta data of the subject, they have this format inside of the Json editor
+            // during load, we will take some information from here and put it in gridinfo (rebuild the about).
+            PEMTaskMetaData: {
+                // inside of the editor, authors is an array, in our case this is slightly different because it is a string,
+                // the owner of the task can write what he want there (website ect...)
+                // we will have to move this to userTaskData during loading.
+                authors: additional.about.authors,
+
+                // Normally this is how we set fr ect...
+                language: window.stringsLanguage,
+
+                // The user should be able to select his version during share, for now one, there is no way to display
+                // the version to the user, but we can set it so later if we need it (example if we want to create a
+                // versioning system), then all the subject already have their own version.
+                version: "1.0.0",
+
+                // the license of the subject, during load we will have to move it to the userTaskData and about.
+                license: additional.about.license
+            },
+
+            // We just need to take the gridInfo from the exercise.
+            gridInfos: that.subtask.gridInfos,
+
+            // We juste take the data from the exercise which mean that we are compatible with everything.
+            data: that.subtask.data,
+
+            // We retrieve the saved answer of the user.
+            // with this method, we can get the updated answer (after save) for all the levels.
+            answer: that.subtask.getAnswerObject()
+        };
+
+        // we remove the userTaskData from subTaskToPublish because the one that is present
+        // is not up to date, we have our additional that is up to date and that we have placed inside of metadatas.
+        if (subTaskToPublish.gridInfos.userTaskData) {
+            delete subTaskToPublish.gridInfos.userTaskData;
         }
+
+        var answerType = that.blocklyHelper.isBlockly ? "blockly" : "python";
+
+        // TODO: change with "prod" url
+        var url = "file:///home/nicolas/stage/test/v01/QuickPi/dynamicTestbed/testbed/index.html";
```

Comment:
It's not very generic to have an URL, specific to quick-pi, in interface-mobileFirst. As this system could be used with multiple tasks, there should be a mechanism for the task to give the URL for sharing.